### PR TITLE
Archive (zip) option also found and corrected an HTTP 404 error 

### PR DESCRIPTION
**Pull request by rsjrny**
1) Fixed a 404 error caused when an activity listed does not exist

2) added a --archive option to create and merge downloaded activity files into a zip file.
after new activities are downloaded the zip file is expanded, new activities added so no duplicates exist, and then files are re-zipped.

### CODE REVIEW

**Comment by pe-st** [comment]
Sorry for not responding earlier. It's quite a big change (two distinct features in one pull request), I wasn't able to test it yet. I have some doubts on some details (e.g. is it good practice to have to have a global variable `NOTEXIST` in upper case; I think the convention is rather lower case? Is it good practice at all to use a global variable here? And is the archiving/zipping part portable? There seem to be hardcoded Windows backslashes?)

File: gcexport.py [hunk 1]
```diff
@@ -97,6 +100,8 @@
 
 MAX_TRIES = 3
 
+NOTEXIST = 0
```
Comment (bxsx) [inline gcexport.py:103]: This looks to be a variable, not a constant. Global variables shall be avoided.

File: gcexport.py [hunk 2]
```diff
@@ -843,10 +855,75 @@ def logging_verbosity(verbosity):
...
+    Let's unzip the archive file first, if it exists,  into a temp area. Now copy
+    the downloaded files into the temp area. This will allow for changed activities to
+    replace the older copy. Then we rebuild the archive. This is done to prevent duplicate
+    file when we re-zip.
+
+    """
+    # data_filename = os.path.join(directory, prefix + 'activity_' + activity_id + append_desc + '.zip')
+    logging.debug("In zipfilesindir, preparing to archive all file in " + src)
+    # sleep to allow system to finish processing file. We don't want an inuse or not found error
+    time.sleep(3)
```
Comment (bxsx) [inline gcexport.py:873]: This is not reliable. You should use a more secure synchronization method.

File: gcexport.py [hunk 3]
```diff
@@ -843,10 +855,75 @@ def logging_verbosity(verbosity):
...
+    dirpart = os.path.dirname(dst)
+    temparch = dirpart + "\\temparch"
```
Comment (bxsx) [inline gcexport.py:875]: It doesn't look cross-platform. You should probably use `os.path.join()` instead.

**Review by bxsx — CHANGES_REQUESTED** [review]
I think this feature is beyond the scope of the app. While unzipping is required for the correct handling of _original_ files, further files processing should be done using dedicated software.

Adding this kind of functionality does not add much value to the application, but it can significantly increase maintenance costs.

However, in case we would like to merge this PR, there are at least a few issues that need to be addressed (after a quick change scan):
- global variables should be omitted
- the code must be cross-platform
- synchronization must be refactored (`time.sleep()`)

In general, I vote against this feature/PR.

**Pull request closed without merging**